Popups and error messages during initial startup of openHAB 5.3 snapshot in Windows 10

When running `start.bat` I get this popup after a while:

<img width="384" height="217" alt="Image" src="https://github.com/user-attachments/assets/a803b288-7aeb-46b7-b3b7-ca790fed7ed3" />

"Den angivne fil blev ikke fundet" translates to something like "The specified file was not found".

I suspect this might be a regression of #1922, since I haven't updated my openHAB development system for a while, and this mentioned "runhidden.vbs" seems to have been added by that PR:

https://github.com/openhab/openhab-distro/blob/b04095e08998763f43d6cd286fe634409cd474c6/distributions/openhab/src/main/resources/bin/karaf.bat#L107

On second run, the situation changed:
```text
Launching the openHAB runtime...

                           _   _     _     ____
   ___   ___   ___   ___  | | | |   / \   | __ )
  / _ \ / _ \ / _ \ / _ \ | |_| |  / _ \  |  _ \
 | (_) | (_) |  __/| | | ||  _  | / ___ \ | |_) )
  \___/|  __/ \___/|_| |_||_| |_|/_/   \_\|____/
       |_|       5.3.0-SNAPSHOT - Build #5535

Use '<tab>' for a list of available commands
and '[cmd] --help' for help on a specific command.
To exit, use '<ctrl-d>' or 'logout'.

openhab> Error executing command: Terminal has been closed
openhab>
```

On one hand, it might have just been a small glitch to be ignored. On the other hand, isn't this message "Error executing command: Terminal has been closed" what should have been fixed by that PR?

> [@jlaur](https://github.com/jlaur) I made some more changes in [#1945](https://github.com/openhab/openhab-distro/pull/1945) to both `karaf.bat` and `runhidden.vbs` so it should work with path names with or without spaces. It works for me on both such paths, so I wonder if you can confirm it also for yourself?

It seems to work better now. I don't know what is causing those other errors:
```text
Launching the openHAB runtime...
Refreshing the cache...
Starting server...
Wait timed out
Finalizing startup sequence...
Error: Could not find or load main class ^
Caused by: java.lang.ClassNotFoundException: ^
'org.apache.karaf.main.Main' is not recognized as an internal or external command,
operable program or batch file.
Restarting JVM...
Error: Could not find or load main class ^
Caused by: java.lang.ClassNotFoundException: ^
'org.apache.karaf.main.Main' is not recognized as an internal or external command,
operable program or batch file.
Restarting JVM...
Error: Could not find or load main class ^
Caused by: java.lang.ClassNotFoundException: ^
'org.apache.karaf.main.Main' is not recognized as an internal or external command,
operable program or batch file.
Restarting JVM...

                           _   _     _     ____
   ___   ___   ___   ___  | | | |   / \   | __ )
  / _ \ / _ \ / _ \ / _ \ | |_| |  / _ \  |  _ \
 | (_) | (_) |  __/| | | ||  _  | / ___ \ | |_) )
  \___/|  __/ \___/|_| |_||_| |_|/_/   \_\|____/
       |_|       5.3.0-SNAPSHOT - Build #5535

Use '<tab>' for a list of available commands
and '[cmd] --help' for help on a specific command.
To exit, use '<ctrl-d>' or 'logout'.

openhab>
```

I'll try again tonight while making sure my `karaf.bat` is fully correct.

--- a/distributions/openhab/src/main/resources/bin/karaf
+++ b/distributions/openhab/src/main/resources/bin/karaf
@@ -414,7 +414,7 @@ awaitOsgiCacheReady() {
         fi
         sleep 1
         SECONDS_WAITED=$((SECONDS_WAITED + 1))
-        if [ "${SECONDS_WAITED}" -ge 60 ]; then
+        if [ "${SECONDS_WAITED}" -ge 70 ]; then
             echo "Wait timed out!"
             break
         fi
@@ -433,7 +433,7 @@ awaitOsgiCacheReady() {
         fi
         sleep 1
         SECONDS_WAITED=$((SECONDS_WAITED + 1))
-        if [ "${SECONDS_WAITED}" -ge 30 ]; then
+        if [ "${SECONDS_WAITED}" -ge 70 ]; then
             echo "Wait timed out!"
             break
         fi

--- a/distributions/openhab/src/main/resources/bin/karaf.bat
+++ b/distributions/openhab/src/main/resources/bin/karaf.bat
@@ -126,7 +126,11 @@ if defined LEVEL (
     )
 )
 if !SECONDS_WAITED! GEQ 30 (
-    echo Wait timed out!
+    if not exist "%STARTLEVEL_FILE%" (
+        echo Wait timed out! .. missing start-level file %STARTLEVEL_FILE%
+        goto SERVER_STARTED
+    )
+    echo Wait timed out! .. did not reach start-level %STARTED_TARGET_LEVEL%
     goto SERVER_STARTED
 )
 timeout /t 1 /nobreak >nul
@@ -152,7 +156,11 @@ if defined LEVEL (
     )
 )
 if !SECONDS_WAITED! GEQ 30 (
-    echo Wait timed out!
+    if not exist "%STARTLEVEL_FILE%" (
+        echo Wait timed out! .. missing start-level file %STARTLEVEL_FILE%
+        goto SERVER_STOPPED
+    )
+    echo Wait timed out! .. did not reach start-level %STOPPED_TARGET_LEVEL%
     goto SERVER_STOPPED
 )
 timeout /t 1 /nobreak >nul

--- a/distributions/openhab/src/main/resources/bin/karaf.bat
+++ b/distributions/openhab/src/main/resources/bin/karaf.bat
@@ -93,6 +93,11 @@ goto AFTER_CACHE_READY
 set CACHE_FOLDER=%KARAF_DATA%\cache\org.eclipse.osgi
 if exist "%CACHE_FOLDER%\.*" goto AFTER_CACHE_READY
 
+if not exist "%DIRNAME%runhidden.vbs" (
+    call :warn Missing helper script: "%DIRNAME%runhidden.vbs"
+    goto END
+)
+
 set CACHE_REFRESH_LOCK=%KARAF_DATA%\cache-refresh-lock
 if exist "%CACHE_REFRESH_LOCK%" goto AFTER_CACHE_READY
 type nul > "%CACHE_REFRESH_LOCK%"
@@ -105,14 +110,9 @@ set STOPPED_TARGET_LEVEL=0
 
 set RUNHIDDEN=wscript //nologo "%DIRNAME%runhidden.vbs"
 
-if not exist "%DIRNAME%runhidden.vbs" (
-    call :warn Missing helper script: "%DIRNAME%runhidden.vbs"
-    goto END
-)
-
 echo Starting server...
 set KARAF_HOME=
-%RUNHIDDEN% """%~dp0karaf.bat"" server"
+%RUNHIDDEN% "%DIRNAME%karaf.bat" "server"
 setlocal EnableDelayedExpansion
 set SECONDS_WAITED=0
 
@@ -138,7 +138,7 @@ endlocal
 
 echo Finalizing startup sequence...
 set KARAF_HOME=
-%RUNHIDDEN% """%~dp0karaf.bat"" stop"
+%RUNHIDDEN% "%DIRNAME%karaf.bat" "stop"
 setlocal EnableDelayedExpansion
 set SECONDS_WAITED=0
 

--- a/distributions/openhab/src/main/resources/bin/karaf.bat
+++ b/distributions/openhab/src/main/resources/bin/karaf.bat
@@ -125,7 +125,7 @@ if defined LEVEL (
         if !LEVEL_A! GEQ %STARTED_TARGET_LEVEL% goto SERVER_STARTED
     )
 )
-if !SECONDS_WAITED! GEQ 30 (
+if !SECONDS_WAITED! GEQ 70 (
     if not exist "%STARTLEVEL_FILE%" (
         echo Wait timed out! .. missing start-level file %STARTLEVEL_FILE%
         goto SERVER_STARTED
@@ -155,7 +155,7 @@ if defined LEVEL (
         if !LEVEL_A! EQU %STOPPED_TARGET_LEVEL% goto SERVER_STOPPED
     )
 )
-if !SECONDS_WAITED! GEQ 30 (
+if !SECONDS_WAITED! GEQ 70 (
     if not exist "%STARTLEVEL_FILE%" (
         echo Wait timed out! .. missing start-level file %STARTLEVEL_FILE%
         goto SERVER_STOPPED

--- a/distributions/openhab/src/main/resources/bin/karaf.bat
+++ b/distributions/openhab/src/main/resources/bin/karaf.bat
@@ -103,8 +103,12 @@ set STARTED_TARGET_LEVEL=10
 set STOPPED_TARGET_LEVEL=0
 > "%STARTLEVEL_FILE%" echo %STOPPED_TARGET_LEVEL%
 
-set RUNHIDDEN=wscript //nologo "%TEMP%\runhidden.vbs"
-> "%TEMP%\runhidden.vbs" echo CreateObject("Wscript.Shell").Run WScript.Arguments(0), 0, False
+set RUNHIDDEN=wscript //nologo "%DIRNAME%runhidden.vbs"
+
+if not exist "%DIRNAME%runhidden.vbs" (
+    call :warn Missing helper script: "%DIRNAME%runhidden.vbs"
+    goto END
+)
 
 echo Starting server...
 set KARAF_HOME=

--- a/distributions/openhab/src/main/resources/bin/runhidden.vbs
+++ b/distributions/openhab/src/main/resources/bin/runhidden.vbs
@@ -1,0 +1,1 @@
+CreateObject("Wscript.Shell").Run WScript.Arguments(0), 0, False

--- a/distributions/openhab/src/main/resources/bin/runhidden.vbs
+++ b/distributions/openhab/src/main/resources/bin/runhidden.vbs
@@ -1,1 +1,1 @@
-CreateObject("Wscript.Shell").Run WScript.Arguments(0), 0, False
+CreateObject("WScript.Shell").Run Chr(34) & WScript.Arguments(0) & Chr(34) & Chr(32) & WScript.Arguments(1), 0, False

